### ISSUE TITLE
make org delete scratch handle expired orgs

### ISSUE THREAD

**Issue by mshanemc** [forcedotcom/cli#2045]
If I have a scratch org that expires, why is it an error if I execute `org delete scratch`?

If I execute `sfdx org delete scratch` , I get the following error:
```Error (1): Attempting to delete an expired or deleted org```
If I execute `sf org delete scratch` , then I get the following error:
```Error (1): Parsing --target-org 
        No authorization information found for <mailto:test-fdyjlnlamife@example.com|test-fdyjlnlamife@example.com>.
See more help with --help```
The old behavior is that it would simply clean up the reference and move on.

And that behavior seems inconsistent with what happens when you execute `sf org list --all --clean` which does simply clean up the reference to the expired scratch org.

[Slack Message](https://salesforce-external.slack.com/archives/C02R1NEG2G3/p1680867956695259?thread_ts=1680867956.695259&cid=C02R1NEG2G3)

**Comment by Julian88Tex** [comment]
Not sure if this is the same or a different issue. I'm getting a different error now on version `@salesforce/cli/1.70.0 darwin-x64 node-v18.14.2`

```
Error (1): Parsing --target-org 
        The org cannot be found
See more help with --help

Try this:

Verify that the org still exists',
If your org is newly created, wait a minute and run your command again',
If you deployed or updated the org's My Domain, logout from the CLI and authenticate again",
If you are running in a CI environment with a DNS that blocks external IPs, try setting SFDX_DISABLE_DNS_CHECK=true'
```

**Comment by WillieRuemmele** [comment]
Hey @Julian88Tex - I believe this should've been fixed in 2.6.10 of plugin-org, which is part of the `latest-rc` version of `sf`, `@salesforce/cli/1.74.7 darwin-arm64 node-v18.7.0`

**Comment by Julian88Tex** [comment]
@WillieRuemmele thanks! For some reason `sf update` seemed to think I was updated but I wasn't. Uninstall and reinstall did the trick.

Was able to git rid of all orgs with connection status `DomainNotFoundError` but having issues with no connected status or `RefreshTokenAuthError` orgs.

Errors I receive:

For `RefreshTokenAuthError` attempt to `sf org delete scratch -o <alias>`

```
Error (1): Unable to associate this scratch org with a DevHub.
```

For `RefreshTokenAuthError` attempt to `sf org delete scratch -o <username>`

```
Error (1): The Dev Hub org cannot be deleted.
```




For empty connected status attempt to `sf org delete scratch -o <alias>`

```
Error (1): Attempting to delete an expired or deleted org
```

For empty connected status attempt to `sf org delete scratch -o <alias>`

```
Error (1): No authorization information found for test-zdnah3odrfgj@example.com.
```

Also tried using `sandbox` instead with each command and now luck.

`sf org list --clean` doesn't hide any of these orgs.
